File: dispatch_event/src/main/java/com/zqg/dispatch_event/MainActivity.java
package com.zqg.dispatch_event;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scoller);
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent event) {
        Log.d(TAG, "dispatchTouchEvent " +  event.toString());
        boolean result = super.dispatchTouchEvent(event);
        Log.d(TAG, "dispatchTouchEvent result " +  result);
        return result;
    }

}

File: sohotfix/src/main/java/com/zqg/sohotfix/MainActivity.kt
package com.zqg.sohotfix

import android.Manifest
import android.content.Context
import android.os.Bundle
import android.os.Environment
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import java.io.File

class MainActivity : AppCompatActivity() {
    private val TAG = javaClass.simpleName
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        requestPermissions(arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE), 100)
    }

    fun dynamic(view: View) {
        val file = File(Environment.getExternalStorageDirectory(), "so/libmain.so")
        val libSoPath = File(externalCacheDir, "so")
        Log.d(TAG, libSoPath.absolutePath)
        if (!libSoPath.exists()) {
            libSoPath.mkdirs()
        }
        val libSoFile = File(libSoPath, "libmain.so")
        FileUtil.copyFile(file, libSoFile)
        Log.d(TAG, libSoFile.absolutePath)
        val soFix = SoFix(this)
        soFix.inject(libSoPath.absolutePath)
    }
    fun loadSo(view: View) {
        val gameSDK = GameSDK()
    }
}
